cornac/models/hrdr/recom_hrdr.py

```diff
@@ -0,0 +1,319 @@
+        self.n_factors = n_factors
+        self.attention_size = attention_size
+        self.n_filters = n_filters
+        self.kernel_sizes = kernel_sizes
+        self.n_user_mlp_factors = n_user_mlp_factors
+        self.n_item_mlp_factors = n_item_mlp_factors
+        self.dropout_rate = dropout_rate
+        self.max_text_length = max_text_length
+        self.max_num_review = max_num_review
+        self.batch_size = batch_size
+        self.max_iter = max_iter
+        self.optimizer = optimizer
+        self.learning_rate = learning_rate
+        self.model_selection = model_selection
+        self.user_based = user_based
+        # Init params if provided
+        self.init_params = {} if init_params is None else init_params
+        self.losses = {"train_losses": [], "val_losses": []}
+
+    def fit(self, train_set, val_set=None):
+        """Fit the model to observations.
+
+        Parameters
+        ----------
+        train_set: :obj:`cornac.data.Dataset`, required
+            User-Item preference data as well as additional modalities.
+
+        val_set: :obj:`cornac.data.Dataset`, optional, default: None
+            User-Item preference data for model selection purposes (e.g., early stopping).
+
+        Returns
+        -------
+        self : object
+        """
+        Recommender.fit(self, train_set, val_set)
+
+        if self.trainable:
+            if not hasattr(self, "model"):
+                from .hrdr import Model
+                self.model = Model(
+                    self.train_set.num_users,
+                    self.train_set.num_items,
+                    self.train_set.review_text.vocab,
+                    self.train_set.global_mean,
+                    n_factors=self.n_factors,
+                    embedding_size=self.embedding_size,
+                    id_embedding_size=self.id_embedding_size,
+                    attention_size=self.attention_size,
+                    kernel_sizes=self.kernel_sizes,
+                    n_filters=self.n_filters,
+                    n_user_mlp_factors=self.n_user_mlp_factors,
+                    n_item_mlp_factors=self.n_item_mlp_factors,
+                    dropout_rate=self.dropout_rate,
+                    max_text_length=self.max_text_length,
+                    pretrained_word_embeddings=self.init_params.get('pretrained_word_embeddings'),
+                    verbose=self.verbose,
+                    seed=self.seed,
+                )
+            self._fit()
+
+        return self
+
+    def _fit(self):
+        import tensorflow as tf
+        from tensorflow import keras
+        from .hrdr import get_data
+        from ...eval_methods.base_method import rating_eval
+        from ...metrics import MSE
+        loss = keras.losses.MeanSquaredError()
+        if not hasattr(self, 'optimizer_'):
+            if self.optimizer == 'rmsprop':
+                self.optimizer_ = keras.optimizers.RMSprop(learning_rate=self.learning_rate)
```

Comment:
For private variable, we usually put underscore before instead of after the name. Some reference: https://google.github.io/styleguide/pyguide.html